tests/integration/test_contact_template.py

```
@@ -1015,5 +1015,260 @@ def test_contact_page_generation_creates_vanity_sort_variants_when_validated(sel
+    def setUp(self):
+        import os
+        from jinja2 import Environment, FileSystemLoader
+        template_dir = os.path.join(
+            os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))),
+            'allium', 'templates'
+        )
+        self.jinja_env = Environment(
+            loader=FileSystemLoader(template_dir), autoescape=True,
+        )
+
+    def _render_pills(self, validation_summary):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_render_pills`

(ANN202)